proto/config/edge_node_cluster.proto

```diff
@@ -0,0 +1,37 @@
+// Copyright(c) 2024 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+package org.lfedge.eve.config;
+option go_package = "github.com/lf-edge/eve-api/go/config";
+option java_package = "org.lfedge.eve.config";
+
+message EdgeNodeCluster {
+  // cluster name, in case it has multiple cluster on the same site
+  string ClusterName = 1;
+
+  // cluster UUID
+  string ClusterID = 2;
+
+  // Cluster-Interface
+  string ClusterInterface = 3;
+
+  // the 'cluster-prefix' IP address of the 'Cluster-Interface', 10.244.244.2/28
+  string ClusterIPPrefix = 4;
+
+  // This device is 'seed-server'
+  bool IsSeedServer = 5;
+
+  // This device is an 'Agent' node
+  bool IsAgent = 6;
+
+  // SSIP = Seed-ServerIP address to join during bootstrap. E.g. 10.244.244.1
+  string SeedServerIP = 7;
+
+  // encrypted token string, use edge-node TPM to decrypt
+  string EncryptedClusterToken = 8;
```

Comment:
This needs to be of type org.lfedge.eve.common.CipherBlock to use the ECDH object encryption.

Reply:
updated